interview-questions.md

```diff
@@ -8,48 +8,104 @@ Then, research the question to expand on your answer. Even if you feel you have
+
+const example = () => {
+    const name = 'CJ';
+    return (
+        <h1> Hello {name}! How are you today?</h1>
+    )
+}
+
+As we see in our above example, we have a JS function with HTML markup inside of it, and in the inner html of our h1 tag, we are pulling the 'name' variable into that inner html with our curly brackets {}.
+(sources - https://beta.reactjs.org/learn/writing-markup-with-jsx, https://beta.reactjs.org/learn/javascript-in-jsx-with-curly-braces)
+
+output// Hello CJ! How are you today?
 
-Researched answer:
 
 1. What is yarn? What file(s) are modified when you run the command yarn in your terminal?
 
-Your answer:
+Your answer: Yarn stands for Yet Another Resource Negotiator, it is a package manager that modifies/adds a few files into a repo it is run in. Those files are package.json, yarn.lock, and node_modules. These files allow us to install things like Jest into a repository and help keep our version of that software up to date.
 
-Researched answer:
+Researched answer: Yarn is a JavaScript package manager developed by a collaboration of Facebook, Exponent, Google, and Tilde to be an open source method to share code with other developers. This code is shared through a package pushed with git version control. When we run yarn in our terminal, we create new files in the repo we are working in based on what has been installed on our local. Ours installs a folder called node_modules, package.json, and yarn.lock. node_modules is our package which contains the dependencies being shared. package.json is a file that generates in our repo that displays the dependencies added with yarn, and yarn.lock is a dependencies graph that describes everything we have installed when running $yarn install.
+(sources - https://yarnpkg.com/getting-started, https://www.arahansen.com/the-ultimate-guide-to-yarn-lock-lockfiles/,https://www.w3resource.com/yarn/creating-a-package.php, https://nodejs.org/en/about/)
 
 3. What is a React component?
 
-Your answer:
+Your answer: Components are what make up a React web application, there are data components which are primarily just showing what the user sees on the page, and then logic components that hold all of the JS logic that react with DOM events. Components can be called upon in our App.js by using component calls (<ComponentName />). Components can have data passed through to them via react props, which effectively act as function parameters in JS. I find it helpful to think of components as puzzle pieces, where the puzzle is the app we are trying to create.
 
```

Comment:
I like this analogy and one of the really cool things is that components can be reused as many times as needed so they become like shape-shifting puzzle pieces that can be plugged in where necessary.